SalesStatsUtility/SalesStatsUtility/src/main/java/StatisticsExporter.java

```diff
@@ -0,0 +1,54 @@
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * StatisticsExporter class is used to generate the output file and to write the statistics
+ */
+public class StatisticsExporter {
+    /**
+     * @param salesStatistic A SalesStatistic object that will provide the results that have to be written
+     * @throws IOException
+     */
+    public void writeStatsToFile(SalesStatistic salesStatistic) throws IOException {
+        File f = new File("salesStatistics.txt");
+        if (f.exists()) {
+            f.delete();
+        } else {
+            f.createNewFile();
+        }
+        String nl = System.getProperty("line.separator");
+        try (BufferedWriter bw = new BufferedWriter(new FileWriter(f))) {
+            List<Product> products = salesStatistic.getProductStatistics(salesStatistic.getProductsHashMap());
```

Comment:
This is not correct design. You are obtaining an encapsulated data structure from the stats object and you pass it back.
The correct way would be: salesStatistic.getProductStatistics()